fuel/utils/write_lock.py

```diff
@@ -0,0 +1,327 @@
+import atexit
```

Comment:
Do we really want to copy all this code to Fuel? Can we not use another library to lock files?